### ISSUE TITLE
Aligning timescale bars to the start of period?

### ISSUE THREAD

**Issue by adam-antonik** [leeoniya/uPlot#310]
I very much like this library, but can't see how to solve one thing that annoys me.

If I plot say a timeseries of 12000 daily values, on the x-axis I have a number of years labeled, with grey grid lines, but these labels and grid lines do not correspond to the start of the year, e.g. the grid line labelled 1978 is at the value for 1978-11-08, similarly if I zoom in, then I get labelled month grid lines, these sit somewhere in the middle of the month.

This makes reading the chart much trickier than it should be.  Is there some option I haven't noticed to align these how I want?

**Comment by leeoniya** [comment]
odd. i have not seen this type of misalignment. all temporal demos which i have seem to work properly. e.g. https://leeoniya.github.io/uPlot/demos/months.html

please create a jsfiddle or provide your data and chart opts which reproduce this issue.

**Comment by adam-antonik** [comment]
https://jsfiddle.net/#&togetherjs=IlsKeGR5QS

Here the 1996 grid line is sometime in December 1996

**Comment by leeoniya** [comment]
that link does not work :(

**Comment by adam-antonik** [comment]
Sorry about that, let me try again https://jsfiddle.net/buyzhq76/

**Comment by leeoniya** [comment]
might have something to do with leap year accounting. seems to start out fine and then goes off the rails.

**Comment by leeoniya** [comment]
ok, so there were a couple things going on here.

with `distr: 2` enabled, there is no knowledge of time (discussed at length in https://github.com/leeoniya/uPlot/issues/311#issuecomment-699748949). all the values are simply treated as labels and uPlot selects equal multiple of that array to display, starting from index 0. the reason it was showing the markers in the middle of the year is simply because the time formatters at multi-year resolutions do not include the more granular parts of the date, since they expect to land on exact Jan 1 boundaries, but `distr: 2` does not guarantee this. anyways, the solution should have been simply to remove `distr: 2`, but that did not work either.

the second issue was a bug in the way year incrementing was done and how the first tick was getting set. it was both a starting point problem, and also a leap year problem. this has been fixed by the linked commit (though i'm not sure if it's fully bulletproof).

your fiddle now works correctly when you drop `distr: 2`.

**Comment by adam-antonik** [comment]
Thanks, that's very helpful.
